### PR TITLE
Change topic name size from 25 to 50

### LINKED ISSUE #14108: DB topic.name uses varying(25) which makes mirrors of repos with long topics fail
- Gitea version: 1.13.0 (also fails on latest dev)
- Git version: 2.26.2
- Operating system:
  - `docker-compose` with the image `gitea/gitea:1`
- Database:
  - [x] PostgreSQL

## Description

1. Create new migration
2. Select GitHub
3. Clone `https://github.com/survivejs/webpack-book` as a **mirror**
4. Will fail with: 

```
Migrate repository from https://github.com/survivejs/webpack-book failed: pq: value too long for type character varying(25)
```

Can be solved by using `varying(255)` on `topic.name`. Is there a reason it's limited to 25?

### PR DESCRIPTION
**Pull request by lunny**
Fix #14108

### CODE REVIEW

**Comment by zeripath** [comment]
I think we should just make this 255

**Comment by lafriks** [comment]
What is the max size for github?

File: models/migrations/v163.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	if err := x.Sync2(new(Topic)); err != nil {
+		return err
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+	if err := recreateTable(sess, new(Topic)); err != nil {
```
Comment (6543) [inline models/migrations/v163.go:29]: **// WARNING: YOU MUST PROVIDE THE FULL BEAN DEFINITION**

Comment (zeripath) [reply models/migrations/v163.go:29]: the `type Topic struct` has to be a full representation of Topic as you need from this point onwards.

**Comment by 6543** [comment]
@lunny created backportable fix: #14152

this pull should still go in too :)

**Comment by zeripath** [comment]
> What is the max size for github?

I'm not sure that there is one. I got it up to 100 characters without a problem. Let me take a look at pathological.

**Review by zeripath — APPROVED** [review]
Although I suspect that we could easily get away with 255 here - 50 appears to be enough to cope with Github

**Review by adelowo — APPROVED** [review]

**Pull request merged**